### PR TITLE
09-Testing-Model TLaS

### PR DESCRIPTION
**Pull request by Noffica**
TLaS complete.

### CODE REVIEW

File: think-like-senior/09-Testing-Model.md [hunk 1]
```diff
@@ -1,11 +1,12 @@
...
 
 ## Start thinking like a senior
-When can we skip testing in a project? Is skipping tests a from of technical debt? How could we spot the difference between debt and bad code? How does testing impact the design of our applications? What issues do slow tests cause?
+When can we skip testing in a project? Is skipping tests a form of technical debt? How could we spot the difference between debt and bad code? How does testing impact the design of our applications? What issues do slow tests cause?
 
-1. Add Validations to your model
-1. Create tests for both success and failure cases. Ensure your model has 100% test coverage.
-1. Create PRs for your work and get your code reviewed.
-1. Review 2 other PRs
+Testing may be skipped when only boilerplate code is present. Another justifiable scenario is when a proof-of-concept (PoC) is being developed. Sometimes, in reality, there are unfortunate cases when PoC's turn into production-grade projects where tests are never added. Such cases would amount to "technical debt".
```
Comment (misterhtmlcss) [inline think-like-senior/09-Testing-Model.md:6]: Here: https://youtu.be/SfWCRl75Kas?t=613

Here: https://youtu.be/SfWCRl75Kas?t=910

TLDR. Technical debt is still good code; it's code we can refactor that is properly written and properly tested.

Can we skip tests in a project? Yes and No. Yes we can whenever we want and no we can't whenever we want. The reality is more akin to Schrödinger's famous cat experiment and understanding whether it's Yes or No highly depends on the constraints of the system which you are writing code in. Are you writing for yourself? Do you like a messy bed? Will others see this code? Is this code important? and so on and on....

Yes you can skip testing.

File: think-like-senior/09-Testing-Model.md [hunk 2]
```diff
@@ -1,11 +1,12 @@
...
 
-**Note**: Some of these test case may require several versions
+"Technical debt" and "bad code" may be differentiated by their nature: the former may be design / architecture decisions expedient for the moment but inefficient for the medium/long -term; the latter may be excessive or convoluted code.
+
+Inclusion of testing would greatly encourage if not force maintainability and comprehensibility. E.g. convoluted relationships amongst different modules, objects etc. would be difficult to test; and vice versa.
+
+If, however, tests are slow (or inefficient in other fundamental aspects) then maintenance of tests becomes de-prioritised by the development team. Tests which fail are ignored or silenced; which only grow in number with time. Eventually, the test suite becomes irrelevant.
```
Comment (misterhtmlcss) [inline think-like-senior/09-Testing-Model.md:12]: Consider when it's acceptable to write or contribute slow tests. Why are the tests slow? IMO odds are good that the tests are poorly written, concealing poorly written code or some other unprofessional conduct that limits the effectiveness of the team and that needs to be addressed pronto.
